cruft/_commands/_generate.py

```diff
@@ -0,0 +1,119 @@
+import json
+import os
+from pathlib import Path
+from shutil import move, rmtree
+from tempfile import TemporaryDirectory
+from typing import Any, Dict, Optional, Set
+
+from cookiecutter.generate import generate_files
+from git import Repo
+
+from .utils import CCContext, generate_cookiecutter_context, get_cruft_file
+
+try:
+    import toml  # type: ignore
+except ImportError:  # pragma: no cover
+    toml = None  # type: ignore
+
+CruftState = Dict[str, Any]
+
+
+def cookiecutter_template(
+    output_dir: Path,
+    repo: Repo,
+    project_dir: Path = Path("."),
+    cookiecutter_input: bool = False,
+    checkout: Optional[str] = None,
+) -> CCContext:
```

Comment:
Can we instead rename this to CookiecutterContext here?

Reply:
Done.